pkg/clamav/session.go

```diff
@@ -312,6 +317,37 @@ func (ses *clamdSession) ScanPath(path string) error {
 	return nil
 }
 
+func (ses *clamdSession) ScanPathCancelable(ctx context.Context, path string) error {
+	walkFn := func(path string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			ses.log(err)
+			return nil
+		}
+
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+
+		if fileInfo.Mode().IsRegular() {
```

Comment:
@mfojtik No. Your snippet is wrong. We must call `log(err)` to append error to report and return `nil` because otherwise, the function will stop walking.